packages/core/src/controllers/events.ts

```diff
@@ -0,0 +1,200 @@
+import { generateChildLogger, Logger } from "@walletconnect/logger";
+import { ICore, IEventClient, EventClientTypes } from "@walletconnect/types";
+import { uuidv4 } from "@walletconnect/utils";
+import {
+  CORE_STORAGE_PREFIX,
+  EVENTS_CLIENT_API_URL,
+  EVENTS_STORAGE_CLEANUP_INTERVAL,
+  EVENTS_STORAGE_CONTEXT,
+  EVENTS_STORAGE_VERSION,
+  RELAYER_SDK_VERSION,
+} from "../constants";
+import { HEARTBEAT_EVENTS } from "@walletconnect/heartbeat";
+import { fromMiliseconds } from "@walletconnect/time";
+
+export class EventClient extends IEventClient {
+  public readonly context = EVENTS_STORAGE_CONTEXT;
+  private readonly storagePrefix = CORE_STORAGE_PREFIX;
+  private readonly storageVersion = EVENTS_STORAGE_VERSION;
+
+  private events = new Map<string, EventClientTypes.Event>();
+  private toPersist = false;
```

Comment:
Does this determine if we should persist events or not ? How come we mutate this variable in functions such as `addTrace` and `setError` ?

nit boolean naming should favor `isTerm`, `hasTerm`, `shouldTerm`
```suggestion
  private hasPersistence = false;
```